app/controllers/admin.js

```diff
@@ -0,0 +1,26 @@
+import Controller from '@ember/controller';
+import { computed, observer } from '@ember/object';
+import { inject as service } from '@ember/service';
+
+export default Controller.extend({
+  session: service(),
+  queryParams: ['title'],
+  title: null,
+  search: null,
+
+  filterBooks: computed('title', function() {
+    let title = this.get('title');
+    if (title) {
+      let books = this.store.query('book', {
+        orderBy: 'title',
+        equalTo: title
+      });
+      return books;
+    }
+  }),
+
+  triggerSearch: observer('search', function() {
```

Comment:
Nice hack ... try to do the same without `observer` and without `transitionToRoute`.